proposals/nnnn-stream-ontouchevent.md

```diff
@@ -0,0 +1,92 @@
+
+First of all the manager should be able to store the last known permission status of OnTouchEvent in case that the permissions were provided before other conditions were met (e.g. permission update during HMI_NONE).
+
+```objc
+@property (assign, nonatomic) BOOL videoStreamPermissionGranted;
+```
+
+During initialization of the manager, it should add an observer to permission changes. On every change the permission change should be read to identify if OnTouchEvent is granted for HMI_FULL. In this case the permission condition is met and the manager should try to start the video service session.
+
+```objc
+- (void)sdl_permissionsChange:(SDLRPCNotificationNotification *)notification {
+    SDLOnPermissionsChange *permissions = notification.notification;
+    NSArray<SDLPermissionItem *> *items = permissions.permissionItem;
+
+    BOOL touchPermissionGranted = NO;
+
+    for (SDLPermissionItem *item in items) {
+        if ([item.rpcName isEqualToString:@"OnTouchEvent"]) {
+            if ([item.hmiPermissions.allowed containsObject:SDLHMILevelFull]) {
+                touchPermissionGranted = YES;
+                break;
+            }
+        }
+    }
+
+    self.videoStreamPermissionGranted = touchPermissionGranted;
+
+    if (self.isHmiStateVideoStreamCapable && self.videoStreamPermissionGranted) {
+        [self sdl_startVideoSession];
+    } else {
+        [self sdl_stopVideoSession];
+    }
+}
+```
+
+With the newly added flag, every other place that starts or stops the video session should be altered as well, including the following:
+
+```objc
+- (void)startWithProtocol:(SDLProtocol *)protocol {
+     // attempt to start streaming since we may already have necessary conditions met
+-    [self sdl_startVideoSession];
++    if (self.isHmiStateVideoStreamCapable && self.videoStreamPermissionGranted) {
++        [self sdl_startVideoSession];
++    } else {
++        [self sdl_stopVideoSession];
++    }
+}
+
+- (void)sdl_hmiStatusDidChange:(SDLRPCNotificationNotification *)notification {
+    ...
+-    if (self.isHmiStateVideoStreamCapable) {
++    if (self.isHmiStateVideoStreamCapable && self.videoStreamPermissionGranted) {
+         [self sdl_startVideoSession];
+     } else {
+         [self sdl_stopVideoSession];
+     }
+}
+```
+
+Note: The example code should not be considered as the only option to implement this proposal. Other solutions that serve the same feature should also be accepted during implementation.  
+
+## Potential downsides
+
+Potentially this code can have impact to existing head units that don't mind with navigation apps to start the video stream. Those applications would require OnTouchEvent anyways for the app to operate hence they shouldn't see any impact. Navigation applications could be delayed by that. However as head units may return `NAK` in case of an early start service request this proposal should be rather seen as a stability improvement.
```

Comment:
I am a little unclear on your intended meaning here - specifically regarding the first sentence - so please let me know if my suggested revisions incorrectly change what you're trying to say.  Thanks!

```suggestion
Potentially this code can have an impact on existing head units that don't know if navigation apps start the video stream. Those applications would require OnTouchEvent anyway for the app to operate, so they shouldn't see any impact. Navigation applications could be delayed by that. However as head units may return `NAK` in case of an early start service request this proposal should rather be seen as a stability improvement.
```

Reply:
I'm sorry for the ambiguity. Touch permissions are not technically required to start a video stream but we imply both to be necessary for navigation apps. 

If a head unit allows apps to start the video stream but touch isn't granted those apps would be basically broken. Touch events would not be forwarded to the app, therefore it can't react per user input.

Reply:
@kshala-ford thanks for the clarification!  I've revised my suggested change in the comment above.  Please commit this change if it makes sense, or feel free to revise as needed.